ui/apps/platform/src/Containers/ComplianceEnhanced/Schedules/ScanConfigsTablePage.tsx

```diff
@@ -93,10 +99,19 @@ function ScanConfigsTablePage({
         [sortOption, page, perPage]
     );
     const { data: listData, isLoading, error, refetch } = useRestQuery(listQuery);
+    const { complianceReportSnapshots } = useWatchLastSnapshotForComplianceReports(
+        listData?.configurations
+    );
 
     const { alertObj, setAlertObj, clearAlertObj } = useAlert();
 
-    const colSpan = hasWriteAccessForCompliance ? 6 : 5;
+    let colSpan = 5;
+    if (hasWriteAccessForCompliance) {
+        colSpan += 1;
+    }
+    if (isReportJobsEnabled) {
+        colSpan += 1;
+    }
```

Comment:
is this just the number of columns in the table? it would be nice to just check the columns list length

Reply:
Could you explain what you mean by `just check the columns list length`? The logic here is an improvement on the following piece of logic:

 `const colSpan = hasWriteAccessForCompliance ? 6 : 5;`
 
We show `5` columns by default. When we have write access for compliance, we will show `6` columns. Now that we show another column on top of that when the report job workflow is enabled, we will show `7`. Pulling up the rendered content for the list of `Th` and deriving the length from that is possible, but it makes it less readable since pieces of code are disjoint. The original person who wrote the code probably had a similar thought process. I would prefer it this way as well.